### ISSUE TITLE
Does the elasticsearch connector support aggregations?

### ISSUE THREAD

**Issue by andys8** [akka/alpakka#744]
Is there a way to use alpakka and elasticsearch aggegrations (e.g. to calculate the sum)?

This [example](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-sum-aggregation.html) describes the query:
```
POST /sales/_search?size=0
{
    "query" : {
        "constant_score" : {
            "filter" : {
                "match" : { "type" : "hat" }
            }
        }
    },
    "aggs" : {
        "hat_prices" : { "sum" : { "field" : "price" } }
    }
}
```

[ElasticsearchSource](https://github.com/akka/alpakka/blob/master/elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchSource.scala) has only a few parameters (index, type, query and some configurations). But it looks like there is no way to provide the `"aggs"`. Am I missing something?

Looking at the [tests](https://github.com/akka/alpakka/blob/master/elasticsearch/src/test/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSpec.scala) the query is always a match all query.

It looks like the connector is limited in the way the api is designed. I'm glad for any help. Thanks a lot.

**Comment by ennru** [comment]
You're right. For the time being the Elasticseach connector offers access to just a tiny bit of the Elasticsearch API.
See #591

**Comment by mbknor** [comment]
I guess this is now possible with latest changes merged to master (not yet released):

https://github.com/akka/alpakka/commit/97f7e46fe41340e147115bb2680eb270150ac609#diff-a05dc5ecaacea54fb635b007c4260294R37

**Comment by arunma** [comment]
@mbknor @ennru Was browsing for issues for my first contribution and thought I could add some testcases for this issue.  I've found out that unfortunately, the "aggs" query does not work even with the `searchParams`. Thought I could take a stab at it and ask for a review but unfortunately, I am stuck.

I've raised a Throw-away PR for your review.  Appreciate if you could give some hints and help me move forward.  Please ignore and close the PR if you have already started working on this issue.  My apologies - I should have given a heads-up.
